src/snovault/elasticsearch/indexer.py

```diff
@@ -334,8 +338,99 @@ def _get_nodes(request, indexer_state):
     return this_node, other_node, continue_on, did_timeout
 
 
+class IndexerInfo(LocalStoreClient):
+    """
+    IndexerInfo specific wrapper for LocalStoreClient
+    - Stores one state hash with managed states
+    - Stores indexing events in list
+    """
+    state_waiting = ('state_waiting', 'Waiting to call endpoint')
+    state_endpoint_start = ('state_endpoint_start', 'Endpoint started running')
+    state_load_indexing = ('state_load_indexing', 'Endpoint checking for uuids to index')
+    state_run_indexing = ('state_run_indexing', 'Endpoint found uuids and started indexing')
+    state_hash_keys = [
+        'endpoint_start',
+        'endpoint_end',
+        'invalidated',
+        'loop_time',
+        'pg_ip',
+        'remote_indexing',
+        'state',
+        'state_desc',
+    ]
+
+    def __init__(self, local_store=None):
+        super().__init__(redis_db=3, local_store=local_store)
+
+    def update_indexer_info(self, state, update):
+        local_store_state = self.hash_get(INDEX_ENDPOINT_INFO)
+        if not local_store_state: 
+            local_store_state = {
+                key: 'empty'
+                for key in self.state_hash_keys
+            }
+        if update:
+            for key, val in update.items():
+                if key in self.state_hash_keys:
+                    local_store_state[key] = val
+        if not hasattr(self, state):
+            state = f"{state} is not a valid state."
+        local_store_state['state'] = getattr(self, state)[0]
+        local_store_state['state_desc'] = getattr(self, state)[1]
+        self.hash_set(INDEX_ENDPOINT_INFO, local_store_state)
+
+    def get_indexing_events(self, start, stop):
+        events = []
+        event_tags_list = self.list_get(INDEX_EVENT_TAGS, start, stop)
+        for event_tag in event_tags_list:
+            end = self.item_get(event_tag + ':end')
+            invalidated = self.item_get(event_tag + ':invalidated', 'pending')
+            duration = self.item_get(event_tag + ':duration', 'pending')
+            events.append(
+                f"{event_tag}: Indexed {invalidated} uuids in {duration} seconds. Ended at {end}"
+            )
+        return events
+
+
+@view_config(route_name='_indexer_info', request_method='GET')
+def indexer_info(request):
+    events_range = request.params.get("events", '0:10')
+    try:
+        colon = events_range.index(":")
```

Comment:
this is an internal endpoint so this suggestion might not be so important, but this throws an exception if the query string `events` has no ':' 
We should either return a BadRequest or default to '0:10' maybe